### ISSUE TITLE
Handle incoming calls that are busy or offline

### ISSUE THREAD

**Issue by 9mido** [TwilioDevEd/voice-javascript-sdk-quickstart-python#5]
If a browser user has a call in progress, and another outside caller tries calling up the browser user, the outside incoming caller just rings continuously and eventually gets disconnected. 

Instead, the outside incoming caller should enter a queue and have hold music waiting for the browser user to finish their current call. There should also be a spoken message like "The phone number you tried calling is busy. Please hold". 

If the browser user is not available at all meaning their Twilio.Device is not ready or offline the incoming caller should be able to leave a voicemail message to the browser user. The spoken message for this should be "The phone number you tried calling is not available. Please leave a message".

Another idea would be to forward this incoming caller to a different browser user who is available. But this is only if more than one browser user exists. Otherwise maybe forward the incoming call to some other phone number like your actual cell / landline phone number as a backup.

**Comment by bld010** [comment]
This is outside the scope of basic functionality we want to include in the Quickstarts. 

If you want to implement something like this, you have a few options. 

Let's say you want to have a caller enter a queue if the Browser Device is busy:
Set up a new endpoint in `app.py` that is called something like '`/no-answer`. In that endpoint, you can use the `<Record>`, `<Play>` or `<Enqueue>` TwiML verbs. 

Within your `/voice` endpoint in `app.py`, use the [`action` attribute on your `dial`](https://www.twilio.com/docs/voice/twiml/dial#action): `dial = Dial(action="/no-answer")`

The `action` attribute will fire a POST request to the specified URL when the call ends (or in this case, the browser client is busy). 

If the Device is offline, you can also provide a fallback URL in your TwiML App. You can point that to another endpoint you create within this or another app or a Twilio Function or a TwiML bin. That's where you can play a message or prompt a caller to record a message.

**Comment by 9mido** [comment]
Your team should make a set of non-quickstart github libraries in each of the programming languages to make something like what I am trying to express in these issues - one place to make and receive calls through the browser and have the other functionalities that a typical business phone system would have. 

Does Twilio Flex seem to do most of what I am trying to express in these issues? Am I better off with that even though I am a 1 person business but in the future will likely expand? I imagine Flex would be more expensive than doing it through code. What about TaskRouter?

When I have time I will try to implement these things you are saying. I will let you know then if I need more assistance.

**Comment by bld010** [comment]
I'd love to chat with you about what functionalities you'd think would be highest value for customers to see in quickstarts/tutorials. Can you email me at the Twilio address in my profile?

Also, I'm the Developer Educator focused on Voice, so Flex is a bit out of my realm of expertise. I'd suggest reaching out to [Twilio Support](https://support.twilio.com/hc/en-us) (there is a link on the top right called 'Talk to Support'). They may have a better idea of what would be best for your business needs.

**Comment by 9mido** [comment]
Basically it is an all in 1 browser page to handle all your business call needs. I have seen something like this before at a previous company I worked for. Each username had something similar to this github project except it had more features. Basically replace an old fashioned physical business phone device where all business call needs would be done all from one page - put any type of call on hold, park the call, transfer the call to someone else, conferences (with or without entrance pins/passwords), etc. Somehow we gave customers 1 department phone number to call incoming and anyone who was logged in could take the call from that 1 department phone number.
